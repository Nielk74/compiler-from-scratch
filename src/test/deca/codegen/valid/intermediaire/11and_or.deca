{
    int x = 1, y = 2, z = 3;

    if (y > z && x != y || z > x) {
        println(y, " > ", z, " && ", x, " != ", y, " || ", z, " > ", x);
    }

    if (y > z && (x != y || z > x)) {
        println("error1");
    } else {
        println(y, " > ", z, " && (", x, " != ", y, " || ", z, " > ", x, ") is false");
    }

    if (x != y && y < z || z < x) {
        println(x, " != ", y, " && ", y, " < ", z, " || ", z, " < ", x);
    }

    if (x != y || y == z && z < x) {
        println(x, " != ", y, " || ", y, " == ", z, " && ", z, " < ", x);
    }

    if ((x != y || y == z) && z < x) {
        println("error2");
    } else {
        println("(", x, " != ", y, " || ", y, " == ", z, ") && ", z, " < ", x, " is false");
    }

    if (x == y || y < z && z > x) {
        println(x, " == ", y, " || ", y, " < ", z, " && ", z, " > ", x);
    }
}