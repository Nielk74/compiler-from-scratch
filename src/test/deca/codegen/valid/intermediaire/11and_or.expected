2 > 3 && 1 != 2 || 3 > 1
2 > 3 && (1 != 2 || 3 > 1) is false
1 != 2 && 2 < 3 || 3 < 1
1 != 2 || 2 == 3 && 3 < 1
(1 != 2 || 2 == 3) && 3 < 1 is false
1 == 2 || 2 < 3 && 3 > 1
